archivebox/extractors/__init__.py

```diff
@@ -136,24 +128,34 @@ def archive_link(link: Link, overwrite: bool=False, methods: Optional[Iterable[s
 
     return link
 
-
 @enforce_types
-def archive_links(links: List[Link], overwrite: bool=False, methods: Optional[Iterable[str]]=None, out_dir: Optional[str]=None) -> List[Link]:
-    if not links:
+def archive_links(all_links: any, overwrite: bool=False, methods: Optional[Iterable[str]]=None, out_dir: Optional[str]=None) -> List[Link]:
+
+    if type(all_links) is list:
```

Comment:
I tend to prefer `isinstance` over `type(...) is ...`: https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance

In this case, however, I'd explicitly check for the QuerySet case, and fallback to generic python iteration otherwise, that way it still supports generators, tuples, etc.

Reply:
Depends...In this case I don't think it matters much. I changed the check: https://github.com/pirate/ArchiveBox/pull/452/commits/ec48f46f2deecbafb7dc0ed360b9e71a33efc9f5